### PR TITLE
Paper Review

### PR DESCRIPTION
**Pull request by kadrlica**
This PR is to track internal review of the easyaccess paper.

### CODE REVIEW

File: paper/paper.md [hunk 1]
```diff
@@ -30,40 +30,40 @@ bibliography: paper.bib
...
 
-One can also load SQL queries from a file into the database, or run SQL queries inside the `easyaccess` python module in another IDE. Most of the features are also exposed through a Python API and can be run inside a Jupyter [@jupyter] notebook or similar tool along with the scientific code analysis.
+One can also load SQL queries from a file into the database, or run SQL queries inside the `easyaccess` python module in another IDE. Most of the features are also exposed through a Python API and can be run inside a Jupyter [@jupyter] notebook or similar tool alongside scientific analyses.
 
-While also using `easyaccess`, users can submit and request cutouts around specific positions or objects which are generated from the images. This allows better integration with other data services for a richer scientific workflow.
+Users can also use `easyaccess` to submit and request cutouts around specific positions or objects which are generated from the images. This allows better integration with other data services for a richer scientific workflow.
 
 ## Architecture
```
Comment (kadrlica) [inline paper/paper.md:64]: Is it correct to say that `easy_or` inherits methods from subclasses, or should it be that subclasses inherit from `easy_or` (or that `easy_or` is a subclass that inherits from a base class)?

File: paper/paper.md [hunk 2]
```diff
@@ -30,40 +30,40 @@ bibliography: paper.bib
...
+ In addition, there are a variety of customized functions to search and describe the tables, search for users and user tables, check quota usage, check the Oracle execution plan, and soon the ability to run asynchronous jobs through a dedicated server. There are dozens of other minor features that allow for a seamless experience while exploring and discovering data within the hundreds of tables inside the DB.
 
-One can also load SQL queries from a file into the database, or run SQL queries inside the `easyaccess` python module in another IDE. Most of the features are also exposed through a Python API and can be run inside a Jupyter [@jupyter] notebook or similar tool along with the scientific code analysis.
+One can also load SQL queries from a file into the database, or run SQL queries inside the `easyaccess` python module in another IDE. Most of the features are also exposed through a Python API and can be run inside a Jupyter [@jupyter] notebook or similar tool alongside scientific analyses.
```
Comment (kadrlica) [inline paper/paper.md:61]: Are cutout jobs really dispatched from easyaccess now? If so, that's awesome!

File: paper/paper.md [hunk 3]
```diff
@@ -30,40 +30,40 @@ bibliography: paper.bib
...
+ `easyaccess` has a variety of features including a history of past commands and smart tab auto-completion for commands, functions, columns, users, tables, and paths. Tables can be written directly into comma-separated-value (CSV) or white-space separated text files, FITS [@FITS] files, and HDF5 [@hdf5] files. It provides an iteratation scheme to avoid memory constraints when retrieving large tables. Tables can also be displayed on the command line and most of the formatting is done using `pandas`. Similarly, priviledged users can easily upload tables to the database from any of the file format described above in order to share data with other users. The uploading mechanism is done chunk-wise, allowing large tables to be loaded while keeping memory usage low.
```
Comment (kadrlica) [inline paper/paper.md]: I'm not really sure what was meant by "an iterator is provided", so take a look at my changes.

File: paper/paper.md [hunk 4]
```diff
@@ -89,6 +90,6 @@ To download easyaccess you can follow the provided options or clone the source c
 
 # Acknowledgments
 
-The DES Data Management System is supported by the National Science Foundation under Grant NSF AST 07-15036 and NSF AST 08-13543
+The DES Data Management System is supported by the National Science Foundation under Grant NSF AST 07-15036 and NSF AST 08-13543. 
 
 # References
```
Comment (kadrlica) [inline paper/paper.md:95]: Do we need the standard DES references?

**Review by kadrlica — COMMENTED** [review]
This paper does an excellent job of describing the easyaccess software and its implementation for DES. I believe that it is ready for submission with the fairly minor changes suggested below.

File: paper/paper.md (hunk 2 above)
Comment (mgckind) [reply paper/paper.md:61]: yes, we need to update the server but that's the idea, and also we are adding submitting asynchronous query jobs as well

File: paper/paper.md (hunk 1 above)
Comment (mgckind) [reply paper/paper.md:64]: yes `easy_or1 inherits methods from multiple subclasses

File: paper/paper.bib [hunk 5]
```diff
@@ -164,60 +127,7 @@ @ARTICLE{DES2016
...
-	{Swanson}, M.~E.~C. and {Tarle}, G. and {Thaler}, J. and {Thomas}, D. and
-	{Thomas}, R.~C. and {Troxel}, M.~A. and {Tucker}, D.~L. and
-	{Vikram}, V. and {Vivas}, A.~K. and {Wal ker}, A.~R. and {Wechsler}, R.~H. and
-	{Weller}, J. and {Wester}, W. and {Wolf}, R.~C. and {Wu}, H. and
-	{Yanny}, B. and {Zenteno}, A. and {Zhang}, Y. and {Zuntz}, J.
-	},
+   author = {{DES Collaboration}},
     title = "{The Dark Energy Survey Data Release 1}",
   journal = {ArXiv e-prints},
 archivePrefix = "arXiv",
```
Comment (mgckind) [inline paper/paper.bib:133]: :+1:

File: paper/paper.md [hunk 6]
```diff
@@ -30,50 +30,51 @@ bibliography: paper.bib
 
 # Summary
 
-`easyaccess` is an enhanced command line interpreter and Python package created to facilitate access to astronomical catalogs stored in SQL Databases. It provides a custom interface with custom commands and was specifically designed to access data from the [Dark Energy Survey](https://www.darkenergysurvey.org/) Oracle database, although gerenerally, it can easily be adapted to extend to another survey or SQL database. The package was completely written in [Python](https://www.python.org/) and support customized addition of commands and functionalities.
+`easyaccess` is an enhanced command line interpreter and Python package created to facilitate access to astronomical catalogs stored in SQL Databases. It provides a custom interface with custom commands and was specifically designed to access data from the [Dark Energy Survey](https://www.darkenergysurvey.org/) Oracle database, although it can easily be extended to another survey or SQL database. The package was completely written in [Python](https://www.python.org/) and support customized addition of commands and functionalities.
```
Comment (mgckind) [inline paper/paper.md:33]: :+1:  Thanks

File: paper/paper.md [hunk 7]
```diff
@@ -30,50 +30,51 @@ bibliography: paper.bib
...
+A significant subset of the DES data was recently [made  public](https://des.ncsa.illinois.edu/releases/dr1) [@DR1] and can be accessed through several mechanisms including `easyaccess` and [web interfaces](https://des.ncsa.illinois.edu/easyweb/) that run `easyaccess` as a backend. This public release includes information for almost 400 million astrophysical sources and complementary tables to allow scientific analysis.
```
Comment (mgckind) [inline paper/paper.md:39]: :+1:

File: paper/paper.md [hunk 8]
```diff
@@ -30,50 +30,51 @@ bibliography: paper.bib
...
 
 ## DES users
 
-The first release of `easyaccess` was on February 17th, 2015 and since then, over 300 users have used it to access the DES databases within the DES collaboration as shown in Figure 1. We note that the number of DES accounts is almost 800, but this is considering all users that had an account including those before the first released version. Recently in August 2018, with version 1.4.4 we added support for the public release and since then we have increased the number of public users.
+The first release of `easyaccess` was on February 17th, 2015 and since then, over 300 DES Collaborators have used it to access the DES databases (Figure 1). We note that roughly 800 DES accounts exist, but this includes all database users including those that created accounts before the release of `easyaccess`. In August 2018 (version 1.4.4), we added support for the public DES data release, and since then we have increased the number of public users.
```
Comment (mgckind) [inline paper/paper.md:43]: :+1:

File: paper/paper.md [hunk 9]
```diff
@@ -30,50 +30,51 @@ bibliography: paper.bib
...
 
 ![Number of user since first version](easyaccess_users.png)
 
 # `easyaccess`
 
-`easyaccess` is a command line interpreter which is heavily based in the [`cmd`](https://docs.python.org/3/library/cmd.html) Python core module and `termcolor` [@termcolor], at its core as well other external and open sourced libraries including NumPy [@NumPy], `pandas` [@pandas], `fitsio` [@fitsio] and `h5py` [@h5py] to handle and transform array data coming to or from the DB, `cx_Oracle` [@cxoracle] to handle the Oracle communication and `requests` [@requests] for external URL requests. Figure 2 shows an example of
-the welcome screen as seen as a DES user.
+`easyaccess` is a command line interpreter that is heavily based on `termcolor` [@termcolor] and the [`cmd`](https://docs.python.org/3/library/cmd.html) Python core module. It interfaces with `cx_Oracle` [@cxoracle] to communicate with Oracle, `requests` [@requests] for external URL requests, and other external open source libraries, including NumPy [@NumPy], `pandas` [@pandas], `fitsio` [@fitsio] and `h5py` [@h5py] to handle and transform array data.
```
Comment (mgckind) [inline paper/paper.md:49]: thanks

File: paper/paper.md [hunk 10]
```diff
@@ -30,50 +30,51 @@ bibliography: paper.bib
...
 
- In addition, there are a variety of customized functions to search and describe the tables, search for users and user tables, check for quota, check the Oracle execution plan, and soon the ability to run asynchronous jobs through a dedicated server. There are dozens of other minor features that allow for a seamless experience while exploring and discovering data within the hundreds of tables inside the DB.
+ In addition, there are a variety of customized functions to search and describe the tables, search for users and user tables, check quota usage, check the Oracle execution plan, and soon the ability to run asynchronous jobs through a dedicated server. There are dozens of other minor features that allow for a seamless experience while exploring and discovering data within the hundreds of tables inside the DB.
 
-One can also load SQL queries from a file into the database, or run SQL queries inside the `easyaccess` python module in another IDE. Most of the features are also exposed through a Python API and can be run inside a Jupyter [@jupyter] notebook or similar tool along with the scientific code analysis.
```
Comment (mgckind) [inline paper/paper.md:60]: thanks!

File: paper/paper.md (hunk 4 above)
Comment (mgckind) [reply paper/paper.md:95]: I've added them, they don't need to be in the md file, they are added when the pdf is compiled using paper.bib

**Review by mgckind — APPROVED** [review]

**Comment by mgckind** [comment]
Thanks for all the comments and edits!

**Pull request merged**